### ISSUE TITLE
efibootmgr missing from UEFI installation

### ISSUE THREAD

**Issue by jscholfield** [deadhead420/arch-linux-anywhere#18]
I ran into an issue, investigated it and I believe found pretty specifically what happened. The efibootmgr package didn't get installed and therefore when grub-install ran during the installation it couldn't create the needed entry in the UEFI variables to add the new boot entry.

I checked to insure  that the efivar -l command did return the efi variables before I started. That shows I was booted in UEFI mode. I then completed the installation on my laptop that already had an existing Arch on a partition other that what I am using for this. 

I did have no trouble now with the install allowing me to leave in place the existing swap and $ESP. I want to thank you for getting that fixed so quick Dylan.

I chose the option to jump out to an arch-chroot environment at the end of the installation. I used pacman to install efibootmgr, which also pulled in efivar. I then ran grub-install as shown in the arch wiki and grub-mkconfig to update the grub configuration file. I ran efibootmgr to check and verify that the new boot entry was added, it was.

Rebooted and all was in order. I did install several times to be sure I didn't somehow cause this. I did also find you can see during the install that efibootmgr isn't being installed with the other boot loader related components.

**Comment by deadhead420** [comment]
You must have been using manual partitioning... Currently the way I have it setup is when you're using auto partition efivar -l will run and it it returns '0' that means your computer has uefi support and it will prompt to install with uefi boot. It then sets 'UEFI' to true then later when grub is installed it does 'if "$UEFI" ; then' and thats when efibootmgr is installed and grub is configured for efi boot. However like I said the way I have things setup not efivar -l is only run for auto partition so UEFI can only be set to true when doing auto partition... So if you're using manual partition when it gets to the point where it installs grub to the drive UEFI is set to false so it assumes you're not using efi boot, doesn't install efibootmgr and configures grub like it would normally without efi.... Thanks for bringing up this issue, I will add the option to enable efi boot in the manual partition function. This will allow the UEFI variable to be set to true when using manual partitioning enabling efibootmgr to be installed with grub.

**Comment by deadhead420** [comment]
Am going to close this issue, thanks again! The fix will be in this coming release 2.1.5 within the next day or two.

**Comment by jscholfield** [comment]
I'm glad you figured this out. I didn't see that issue in the code but it makes perfect sense in that it would cause this. Good lesson for me to cover the basics completely when I troubleshoot. I saw all that logic but not that efivar only ran for auto partition. Thanks for the good work.

**Comment by deadhead420** [comment]
Latest release is out and this issue has been 100% addressed. Just installed arch from start to finish with efi boot using manual partitioning. It now prompts to enable efi but only if efivar -l returns 0 (meaning your are already booted into efi on the install media), and if you have an EFI system boot partition already setup. It also checks the filesystem type on the efi partition and if it is not vfat (required) will prompt to format to vfat filesystem. UEFI is then set to true and efibootmgr is installed, as well as grub installed in efi mode. Thanks again for your report and enjoy!

**Comment by jscholfield** [comment]
Thanks much for the quick attention and work on this. I've been watching for the new release and have already tested it. I jumped out using the arch-chroot option and ran efibootmgr to verify the new boot entry. This time efibootmgr did get installed but the entry didn't get created in the UEFI variables. I got interrupted but I'm going to re-install and test further. I did see the new prompts and display that efibootmgr was being installed but I didn't notice anything indicating that grub-mkconfig ran but I could have just missed it. I ran grub-mkconfig and the entry then did get created. I'll be taking a look to see that in the code if it fails my second test but I thought I should pass this on even though I don't yet know all I should about this.
